### DISCUSSION TITLE
Get device and its channels

### DISCUSSION THREAD

**Discussion by zefir-git** [liquidctl/liquidctl#607] [Ideas]
Currently there is no way (known to me) to list a device's channels, other than to error it with an invalid channel (which you have to be careful as technically anything could be valid) and then parse the error message. This is really annoying. A command to get a device and list its channels would IMO be ideal. Additionally, additional information could be accessible per-channel.

Example:
```
$ liquidctl -n 0 get --json
```
```json
{
  "bus": "hid",
  "address": "/dev/hidraw4",
  "description": "Some Device",
  "channels": [
    "fan1": {
       "value": 1700,
       "unit": "RPM",
       "mode": "PWM"
    },
    "fan2": {
      "value": 850,
      "unit": "RPM",
      "mode": "DC"
    }
  ]
}
```

Currently, you can get value, unit and mode information with `initialize`, however, not the channel (e.g. `fan1`, `fan2`) without which you can't effectively use this CLI as an API.

Or perhaps this info could just be available in `… list --json` and `… list -v`.

Or also `… get <channel>` could show the info for the specified channel only.

**Comment by jonasmalacofilho** [comment]
Related: #434
